File: data/reading/reading.go
package reading

import (
	"fmt"
	"time"
)

type Reading struct {
	Sensor

	time.Time

	Temperature
	Humidity
}

func (reading Reading) String() string {
	//formattedTime := reading.Time.Format("02.01.2006. 15:04:05")
	return fmt.Sprintf("%s, %s", reading.Temperature, reading.Humidity)
}

File: Common/Models/Reading.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace TemperatureTracker.Models;

public class Reading {
	[Key]
	[JsonPropertyName("id")]
	public int ReadingId { get; set; }

	[JsonPropertyName("time")]
	public DateTime Time { get; set; } = DateTime.Now;

	[JsonPropertyName("temperature")]
	public double? Temperature { get; set; }

	[JsonPropertyName("humidity")]
	public double? Humidity { get; set; }

	[JsonPropertyName("sensor")]
	public int SensorId { get; set; }

	[JsonIgnore]
	public Sensor Sensor { get; set; } = null!;
}